File: CS/Solution43.Module.Win/Controllers/HideNavigationPanelButtonsController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using DevExpress.ExpressApp;

namespace Solution43.Module.Win.Controllers {
    // For more typical usage scenarios, be sure to check out https://documentation.devexpress.com/eXpressAppFramework/clsDevExpressExpressAppViewControllertopic.aspx.
    public class HideNavigationPanelButtonsController : WindowController {
        protected override void OnActivated() {
            base.OnActivated();
            Frame.TemplateChanged += Frame_TemplateChanged;
        }
        private void Frame_TemplateChanged(object sender, EventArgs e) {
            Form form = (Form)Frame.Template;
            form.Load += Form_Load;
        }
        private void Form_Load(object sender, EventArgs e) {
            if(Frame.Template is DevExpress.ExpressApp.Win.Templates.INavigationPanelHolder) {
                DevExpress.XtraBars.Docking.DockPanel navigationPanel = ((DevExpress.ExpressApp.Win.Templates.INavigationPanelHolder)Frame.Template).DockPanelNavigation;
                navigationPanel.Options.ShowAutoHideButton = false;
                navigationPanel.Options.ShowCloseButton = false;
            }
        }
        protected override void OnDeactivated() {
            Frame.TemplateChanged -= Frame_TemplateChanged;
            base.OnDeactivated();
        }
    }
}

File: VB/Solution43.Module.Win/Controllers/HideNavigationPanelButtonsController.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.ExpressApp

Namespace Solution43.Module.Win.Controllers

    ' For more typical usage scenarios, be sure to check out https://documentation.devexpress.com/eXpressAppFramework/clsDevExpressExpressAppViewControllertopic.aspx.
    Public Class HideNavigationPanelButtonsController
        Inherits WindowController

        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            AddHandler Frame.TemplateChanged, AddressOf Frame_TemplateChanged
        End Sub

        Private Sub Frame_TemplateChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim form As Form = CType(Frame.Template, Form)
            AddHandler form.Load, AddressOf Form_Load
        End Sub

        Private Sub Form_Load(ByVal sender As Object, ByVal e As EventArgs)
            If TypeOf Frame.Template Is DevExpress.ExpressApp.Win.Templates.INavigationPanelHolder Then
                Dim navigationPanel As DevExpress.XtraBars.Docking.DockPanel = CType(Frame.Template, DevExpress.ExpressApp.Win.Templates.INavigationPanelHolder).DockPanelNavigation
                navigationPanel.Options.ShowAutoHideButton = False
                navigationPanel.Options.ShowCloseButton = False
            End If
        End Sub

        Protected Overrides Sub OnDeactivated()
            RemoveHandler Frame.TemplateChanged, AddressOf Frame_TemplateChanged
            MyBase.OnDeactivated()
        End Sub
    End Class
End Namespace
